src/main/java/io/anserini/collection/AclCollection.java

```diff
@@ -0,0 +1,201 @@
+      super(path);
+
+      // read YAML file into JsonNode format
+      bufferedReader = new BufferedReader(new FileReader(path.toString()));
+      ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+      MappingIterator<JsonNode> iterator =
+        mapper.readerFor(JsonNode.class).readValues(bufferedReader);
+
+      if (iterator.hasNext()) {
+        JsonNode root = iterator.next();
+        iter = root.fields();
+        if (iter.hasNext()) {
+          nodeEntry = iter.next();
+        }
+      }
+    }
+
+    @Override
+    public void readNext() throws NoSuchElementException {
+      if (nodeEntry == null) {
+        throw new NoSuchElementException("JsonNode is empty");
+      } else {
+        bufferedRecord = new AclCollection.Document(nodeEntry);
+        if (iter.hasNext()) {
+          nodeEntry = iter.next();
+        } else {
+          atEOF = true; // there is no more JSON object in the bufferedReader
+        }
+      }
+    }
+  }
+
+  /**
+   * A document in a JSON collection.
+   */
+  public class Document implements SourceDocument {
+    private String id;
+    private String contents;
+    private JsonNode paper;
+    private JsonNode volume;
+    private List<String> authors;
+    private List<String> venues;
+    private List<String> sigs;
+
+    public Document(Map.Entry<String, JsonNode> jsonEntry) {
+      id = jsonEntry.getKey();
+      paper = jsonEntry.getValue();
+      contents = getOrDefault(paper, "title") + " " + getOrDefault(paper, "abstract_html");
+
+      // Process author facets
+      authors = new ArrayList<>();
+
+      if (paper.has("author")) {
+        ArrayNode authorNode = (ArrayNode) paper.get("author");
+        authorNode.elements().forEachRemaining(node -> {
+          authors.add(((ObjectNode) node).get("full").asText());
+        });
+      }
+
+      // Retrieve parent volume metadata
+      String parentVolumeId = paper.get("parent_volume_id").asText();
+      volume = AclCollection.this.volumes.get(parentVolumeId);
+
+      // Process venue facets
+      venues = new ArrayList<>();
+      ArrayNode venuesNode = (ArrayNode) volume.get("venues");
+      venuesNode.elements().forEachRemaining(node -> {
+        venues.add(node.asText());
+      });
+
+      // Process SIG facets
+      sigs = new ArrayList<>();
+      ArrayNode sigsNode = (ArrayNode) volume.get("sigs");
+      sigsNode.elements().forEachRemaining(node -> {
+        sigs.add(node.asText());
+      });
+    }
+
+    @Override
+    public String id() {
+      if (id == null) {
+        throw new RuntimeException("JSON document has no \"id\" field");
+      }
+      return id;
+    }
+
+    @Override
+    public String content() {
+      if (contents == null) {
+        throw new RuntimeException("JSON document has no \"contents\" field");
+      }
+      return contents;
+    }
+
+    @Override
+    public boolean indexable() {
+      return true;
+    }
+
+    public JsonNode paper() {
```

Comment:
I'm a bit torn. Your method naming is cleaner IMO, but Java style guidelines actually dictate `getX()`... thoughts?

Reply:
Yeah I thought it'd be more consistent since we already have the `id()` and `content()` functions instead of `getId()` and `getContent() baked into the `Document` class, if we changed it to Java guidelines we'd probably want to change those as well

Reply:
Yup, let's leave as is.